src/main/java/ninjabrainbot/data/endereye/Throw.java

```diff
@@ -56,15 +63,32 @@ public static IThrow parseF3C(String string, double crosshairCorrection, IModifi
 			boolean nether = substrings[2].equals("minecraft:the_nether");
 			double x = Double.parseDouble(substrings[6]);
 			double z = Double.parseDouble(substrings[8]);
-			double alpha = Double.parseDouble(substrings[9]);
+			double rawAlpha = Double.parseDouble(substrings[9]);
 			double beta = Double.parseDouble(substrings[10]);
-			alpha += crosshairCorrection;
-			return new Throw(x, z, alpha, beta, nether, modificationLock);
+			double alpha = getPreciseAlpha(rawAlpha, preferences, dataStateHandler.getDataState().boatAngle().get());
+			return new Throw(x, z, rawAlpha, alpha, beta, nether, dataStateHandler.getModificationLock());
 		} catch (NullPointerException | NumberFormatException e) {
 			return null;
 		}
 	}
 
+	private static double getPreciseAlpha(double alpha, NinjabrainBotPreferences preferences, float boatAngle) {
+		if (preferences.usePreciseAngle.get()) {
+			double sens = preferences.sensitivity.get();
+			double preMult = sens * 0.6f + 0.2f;
+			preMult = preMult * preMult * preMult * 8.0f;
+			double minInc = preMult * 0.15D;
+			alpha = boatAngle + Math.round((alpha - boatAngle) / minInc) * preMult * 0.15D;
+		}
+
+		alpha += preferences.crosshairCorrection.get();
+
+		// Determined experimentally, exact cause unknown
+		alpha -= 0.00079 * Math.sin((alpha + 45) * Math.PI/180.0);
```

Comment:
I dont know if im comfortable adding this if we know so little about it. Ill think about it. What was your method of gathering the 400 data points?

Reply:
Spawn into a random world, throw and measure an eye, then move ~20 blocks to the side and measure another eye (2 throws to guarantee 100% confidence). I repeated this 200 times, after each world printing the angle and error for the 2 throws using a modified ninbot.

You can also check for a particular throw by increasing fov until pixels change, then calculate the error much more accurately using the pixel transition point. I've used this to confirm that the effect is also present without tall res and for different distances.

I expect that we'll eventually get a better understanding of why this happens, and then we can probably optimize the adjustment. But I think the main thing is that the current adjustment is much better than no adjustment.